code/modules/mod/modules/_module.dm

```diff
@@ -346,9 +348,16 @@
 		used_overlay = overlay_state_inactive
 	else
 		return
-	var/mutable_appearance/module_icon = mutable_appearance(overlay_icon_file, used_overlay, layer = standing.layer + 0.1)
+	var/mutable_appearance/module_icon
+	if(mask_worn_overlay)
+		var/icon/mod_mask = icon(mod.generate_suit_mask())
+		mod_mask.Blend(icon(overlay_icon_file, used_overlay), ICON_MULTIPLY)
```

Comment:
for the record I would ideally like to cache this but if it can't be done I understand